src/drivers/lsm303d/lsm303d.cpp

```diff
@@ -626,6 +626,9 @@ LSM303D::init()
 
 	reset();
 
+	/* Prime _mag with parents devid. */
+	_mag->_device_id.devid = _device_id.devid;
```

Comment:
I don't understand this call compared to this one:
https://github.com/PX4/Firmware/pull/1715/files#diff-fc165566877bb4af47d4be96eb8b7690R1769

Should the device ID maybe be set in the _device_id constructor?